src/Popper.js

```diff
@@ -13,7 +14,7 @@ import { unwrapArray, setRef } from './utils';
 import { usePopper } from './usePopper';
 
 type ReferenceElement = ?(VirtualElement | HTMLElement);
-type Modifiers = Array<$Shape<Modifier<any>>>;
+type Modifiers = Array<$Shape<StrictModifiers | Modifier<string, {}>>>;
```

Comment:
I'm not very familiar with Flow's types, so the type here may be incorrect.
What I was going for is the equivalent of this in TypeScript:
```ts
type Modifiers = Array<StrictModifiers | Partial<Modifier<string, object>>>;
```
In simple terms - an array of items, where each item is either `StrictModifiers` (which is a union of types, so one of them), or `Partial<Modifier<string, object>>` - which is an object with the same shape as `Modifier`, in which all properties are optional - the generics passed to it are `string` for `Name` and any object (so `{ foo: 'bar' }` would work, but `true` wouldn't) for `Options` (note that `object` is not the same as the `Object` type).

Reply:
Do you know why autocomplete works but not typechecking? When I hover the tooltip it shows the types, but doesn't actually error for wrong types.

![autocomplete](https://i.imgur.com/dPWXq6n.png)

![no error](https://i.imgur.com/NjoeE2x.png)

Reply:
Okay, it seems like adding some dynamic behavior inside the modifiers array stops TypeScript from working 😅

Reply:
I'm guessing this is a typescript file? Not sure from the screenshot.
Originally my changes weren't able to make the type-checking work properly, since a Modifer can be a Strict one (i.e. `name` is one of `	"offset" | "applyStyles" | "arrow" | "hide" | "computeStyles" | "eventListeners" | "flip" | "preventOverflow" | "popperOffsets"` or an external one (`name` is any string other than the aforementioned ones).

I've just pushed a change that should make the distinction at compile-time, and allow both IDE autocompletion as well as compile-time type checking for the different modifiers - so this wouldn't compile:
```ts
{
  name: 'flip',
  options: {
	altBoundary: 1  // Type 'number' is not assignable to type 'boolean | undefined'.
  }
}
```

and neither would this:
```ts
{
  name: 'offset',
  options: {
	unknownOption: true
  }
}
```

This would though:
```ts
{
  name: 'my-modifier',
  options: {
    someOption: true,
    anotherOption: 'test'
  }
}
```